catalog/spatial/csw/spatial-csw-transformer/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/TestGmdTransformer.java

```diff
@@ -0,0 +1,140 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.TreeSet;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.spatial.ogc.csw.catalog.common.GmdMetacardType;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+
+public class TestGmdTransformer {
+
+    private GmdTransformer gmdTransformer;
+
+    private static final TreeSet<String> SUBJECTS = new TreeSet<>(Arrays.asList("Geologie",
+            "World",
+            "boundaries",
+            "elevation",
+            "inlandWaters",
+            "oceans",
+            "society",
+            "structure",
+            "transportation",
+            "utilitiesCommunication"));
+
+    private void assertGmdMetacard(Metacard metacard) {
+        assertThat(metacard.getMetacardType()
+                .getName(), is(GmdMetacardType.GMD_METACARD_TYPE_NAME));
+    }
+
+    @Before
+    public void setUp() {
+        gmdTransformer = new GmdTransformer();
+    }
+
+    @Test(expected = IOException.class)
+    public void testBadInputStream() throws Exception {
+        InputStream is = Mockito.mock(InputStream.class);
+        doThrow(new IOException()).when(is)
+                .read(any());
+        gmdTransformer.transform(is);
+    }
+
+    @Test
+    public void testDataset() throws Exception {
+        Metacard metacard = transform("/gmd/dataset.xml");
+        assertGmdMetacard(metacard);
+        DateTimeFormatter dateFormatter = ISODateTimeFormat.dateOptionalTimeParser();
+        Date expectedDate = dateFormatter.parseDateTime("2004-03-14")
+                .toDate();
+
+        assertThat(metacard.getId(), is("550e8400-e29b-41d4-a716-48957441234"));
+
+        assertThat(metacard.getContentTypeName(), is("dataset"));
+
+        assertThat(metacard.getAttribute(GmdMetacardType.GMD_TYPE)
+                .getValue(), is("dataset"));
+
+        assertThat(metacard.getModifiedDate(), is(expectedDate));
+        assertThat(metacard.getCreatedDate(), is(expectedDate));
+
+        assertThat(metacard.getAttribute(GmdMetacardType.GMD_CRS)
+                .getValue(), is("urn:ogc:def:crs:World Geodetic System::WGS 84"));
+
+        assertThat(metacard.getTitle(), is("VMAPLV0"));
+        assertThat(metacard.getAttribute(GmdMetacardType.GMD_ABSTRACT)
+                .getValue(), is(
+                "Vector Map: a general purpose database design to support GIS applications"));
+
+        TreeSet<String> subjectAttributes = new TreeSet<>();
+        metacard.getAttribute(GmdMetacardType.GMD_SUBJECT)
+                .getValues()
+                .forEach(subject -> subjectAttributes.add((String) subject));
+        assertThat(SUBJECTS, is(subjectAttributes));
```

Comment:
Nit, but I think the order should be `assertThat(actualValue, is(expectedValue))`.